Mifare.c

```diff
@@ -1,203 +1,406 @@
+    status = phacDiscLoop_SetConfig(&sDiscLoop, PHAC_DISCLOOP_CONFIG_PAS_POLL_TECH_CFG, PHAC_DISCLOOP_POS_BIT_MASK_A);
+    CHECK_STATUS(status);
+
+    /* Turn OFF Passive Listen. */
+    status = phacDiscLoop_SetConfig(&sDiscLoop, PHAC_DISCLOOP_CONFIG_PAS_LIS_TECH_CFG, PH_OFF);
+    CHECK_STATUS(status);
+
+    /* Turn OFF active listen. */
+    status = phacDiscLoop_SetConfig(&sDiscLoop, PHAC_DISCLOOP_CONFIG_ACT_LIS_TECH_CFG, PH_OFF);
+    CHECK_STATUS(status);
+
+    /* Turn OFF Active Poll */
+    status = phacDiscLoop_SetConfig(&sDiscLoop, PHAC_DISCLOOP_CONFIG_ACT_POLL_TECH_CFG, PH_OFF);
+    CHECK_STATUS(status);
+
+    /* Disable LPCD feature. */
+    status = phacDiscLoop_SetConfig(&sDiscLoop, PHAC_DISCLOOP_CONFIG_ENABLE_LPCD, PH_OFF);
+    CHECK_STATUS(status);
+
+    /* reset collision Pending */
+    status = phacDiscLoop_SetConfig(&sDiscLoop, PHAC_DISCLOOP_CONFIG_COLLISION_PENDING, PH_OFF);
+    CHECK_STATUS(status);
+
+    /* whether anti-collision is supported or not. */
+    status = phacDiscLoop_SetConfig(&sDiscLoop, PHAC_DISCLOOP_CONFIG_ANTI_COLL, PH_ON);
+    CHECK_STATUS(status);
+
+    /* Device limit for Type A */
+    status = phacDiscLoop_SetConfig(&sDiscLoop, PHAC_DISCLOOP_CONFIG_TYPEA_DEVICE_LIMIT, PH_ON);
+    CHECK_STATUS(status);
+
+    /* Discovery loop Operation mode */
+    status = phacDiscLoop_SetConfig(&sDiscLoop, PHAC_DISCLOOP_CONFIG_OPE_MODE, RD_LIB_MODE_NFC);
+
+    /* Bailout on Type A detect */
+    status = phacDiscLoop_SetConfig(&sDiscLoop, PHAC_DISCLOOP_CONFIG_BAIL_OUT, PHAC_DISCLOOP_POS_BIT_MASK_A);
+    CHECK_SUCCESS(status);
+
+    /* Return Status */
+    return status;
 }
 
-PyObject *Mifare_select(Mifare *self)
+phStatus_t NfcRdLibInit(void)
 {
-    uint8_t byteBufferSize = UID_BUFFER_SIZE;
-    uint8_t byteBuffer[UID_BUFFER_SIZE];
 
-    uint8_t pAtqa[2];
     phStatus_t status;
 
-    uint8_t bSak;
-    uint8_t bMoreCardsAvailable;
+    /* Initialize the Reader BAL (Bus Abstraction Layer) component */
+    status = phbalReg_Stub_Init(&sBalReader, sizeof(phbalReg_Stub_DataParams_t));
+    CHECK_SUCCESS(status);
+
+    /* Initialize the OSAL Events. */
+    status = phOsal_Event_Init();
+    CHECK_STATUS(status);
+
+    //Start interrupt thread
+    Set_Interrupt();
+
+    /* Set HAL type in BAL */
+    #ifdef NXPBUILD__PHHAL_HW_PN5180
+        status = phbalReg_SetConfig(
+            &sBalReader,
+            PHBAL_REG_CONFIG_HAL_HW_TYPE,
+            PHBAL_REG_HAL_HW_PN5180);
+    #endif
+    #ifdef NXPBUILD__PHHAL_HW_RC523
+        status = phbalReg_SetConfig(
+            &sBalReader,
+            PHBAL_REG_CONFIG_HAL_HW_TYPE,
+            PHBAL_REG_HAL_HW_RC523);
+    #endif
+    #ifdef NXPBUILD__PHHAL_HW_RC663
+        status = phbalReg_SetConfig(
+            &sBalReader,
+            PHBAL_REG_CONFIG_HAL_HW_TYPE,
+            PHBAL_REG_HAL_HW_RC663);
+    #endif
+    CHECK_STATUS(status);
+
+    status = phbalReg_SetPort(
+        &sBalReader,
+        SPI_CONFIG);
```

Comment:
Adding an explicit cast to `uint8_t *` on this line will resolve the warning you're still seeing.